### PR TITLE
fixes issues 11, 13, 14, 15

### LINKED ISSUE #11: URL paths incorrect when using a notebook in a subdirectory
Let's say the notebook server is started in ``~/foo``. I then navigate to ``~/foo/bar``, and open the notebook ``~/foo/bar/example.ipynb``. This starts an ipython kernel with its cwd set to ``~/foo/bar``. If padre now wants to render ``~/foo/bar/baz/image.png``, it forms the URL as ``/files/baz/image.png`` (since it forms the URL relative to its cwd). This is incorrect and fails to render; the correct URL should be relative to the server's cwd, i.e. ``/files/bar/image.png``.

The correct logic is to figure out the notebook server cwd (i.e. the directory that the URL ``/files`` maps to), compare that to the kernel cwd, and adjust URLs appropriately.

``IPython.html.notebookapp.list_running_servers()`` seems to be a way to get at the server cwd. But, what if we have more than one server running? How does a notebook tell which server it was launched under?

### PR DESCRIPTION
**Pull request by o-smirnov**
And all sorts of other cleanups.

### CODE REVIEW

**Comment by o-smirnov** [comment]
OK @gijzelaerr , same problem again here, my master has diverged from the radio-astro master. I still don't have a good workflow for merging your changes back into my repo. Your last suggestion (making a branch in my repo that would track the radio-astro master) didn't work so well. And I can't find a way on github to initiate a pull request *from* radio-astro to my repo, i.e. the opposite way. In fact I can't even display a comparison, the damn GUI simply won't let me!

Given that I'm still actively developing, it makes sense for me to maintain my own repo full time, or a full-time branch. So how about we just create a "devel" branch on radio-astro/radiopadre which I can play on, and I can issue pull requests every now and then?

**Comment by gijzelaerr** [comment]
I have no idea what is going on from here at this point. But I think you are missing things up big time ;)

Looks like you have the old commits that I rebased (squashed) also in your tree. I think because you are not working with a feature branch but are committing directly to your master branch. You should not do that.

Maybe a good idea to just start over and walk through it over chat.

File: radiopadre/__init__.py [hunk 1]
```diff
@@ -35,9 +36,116 @@
...
 astropy.log.setLevel('ERROR')
 
 
+def _init_js_side ():
+    """Checks that Javascript components of radiopadre are initialized"""
+    try:
+        get_ipython
+    except:
+        return None
+    get_ipython().magic("matplotlib inline")
```
Comment (gijzelaerr) [inline radiopadre/__init__.py]: i think it is better to put this in a JS file and import it from there.

but i'm rebasing this now so do that in a later commit

**Comment by gijzelaerr** [comment]
pff, this is such a mess now, merging this is taking too much time. I give up. we should start over

**Comment by o-smirnov** [comment]
OK @gijzelaerr , I managed to merge radiopadre/master into my fork's master, which has resolved all conflicts and has made this pull request automatically merge-able again.

I've bumped the version number to 0.3.

I suggest we merge this one, release it, then I'll create a development branch for myself under here and delete my fork.

**Comment by gijzelaerr** [comment]
yes, it is mergable, but on merging history will be a complete mess. Let me have a look.

**Comment by gijzelaerr** [comment]
rebased into one commmit

https://github.com/radio-astro/radiopadre/commit/e2b18b7eea2be0a94e461206ab292caa34c94703

**Pull request closed without merging**